drivers/metadata/metadata.go

```diff
@@ -831,6 +843,175 @@ func (s Sequence) values() []interface{} {
 	}
 }
 
+type PrivilegeSummarySet struct {
+	resultSet
+}
+
+func NewPrivilegeSummarySet(v []PrivilegeSummary) *PrivilegeSummarySet {
+	r := make([]Result, len(v))
+	for i := range v {
+		r[i] = &v[i]
+	}
+	return &PrivilegeSummarySet{
+		resultSet: resultSet{
+			results: r,
+			columns: []string{
+				"Schema",
+				"Name",
+				"Type",
+				"Access privileges",
+				"Column privileges",
+			},
+		},
+	}
+}
+
+func (s PrivilegeSummarySet) Get() *PrivilegeSummary {
+	return s.results[s.current-1].(*PrivilegeSummary)
+}
+
+// PrivilegeSummary summarizes the privileges granted on a database object
+type PrivilegeSummary struct {
+	Catalog          string
+	Schema           string
+	Name             string
+	ObjectType       string
+	ObjectPrivileges ObjectPrivileges
+	ColumnPrivileges ColumnPrivileges
+}
+
+func (s PrivilegeSummary) values() []interface{} {
+	return []interface{}{
+		s.Catalog,
+		s.Schema,
+		s.Name,
+		s.ObjectType,
+		s.ObjectPrivileges,
+		s.ColumnPrivileges,
+	}
+}
+
+// Privilege granted on database object
+type ObjectPrivilege struct {
+	Grantee       string
+	Grantor       string
+	PrivilegeType string
+	IsGrantable   bool
+}
+
+// Privilege granted on column
+type ColumnPrivilege struct {
+	Column        string
+	Grantee       string
+	Grantor       string
+	PrivilegeType string
+	IsGrantable   bool
+}
+
+// Privileges granted on a database object
+type ObjectPrivileges []ObjectPrivilege
+
+// Privileges granted on a column
+type ColumnPrivileges []ColumnPrivilege
+
+func (p ObjectPrivileges) Len() int      { return len(p) }
+func (p ObjectPrivileges) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+func (p ObjectPrivileges) Less(i, j int) bool {
+	switch {
+	case p[i].Grantee != p[j].Grantee:
+		return p[i].Grantee < p[j].Grantee
+	case p[i].Grantor != p[j].Grantor:
+		return p[i].Grantor < p[j].Grantor
+	}
+	return p[i].PrivilegeType < p[j].PrivilegeType
+}
+
+func (p ObjectPrivileges) String() string {
+	if len(p) == 0 {
+		return ""
+	}
+
+	sort.Sort(p)
```

Comment:
Can this be done once on struct initalization?

Reply:
Right now the privileges objects are extended iteratively. So on object initialization the slice is not yet complete. I wanted to make it easy to build the privileges objects from the query result. If you would have to collect all privileges first into a slice and then initialize the privileges object (which is just a slice anyways currently) that would make the `PrivilegeSummaries` method slightly more complex. And in practice the String method and with it the Sort would anyways only be executed once.

So I guess the current solution favors readability and less complexity over performance. But it could be changed of course.

Reply:
Sure. I removed the sort. In case of the information schema reader the values are anyways sorted by the query already. Other callers could simply sort the privileges themselves. I left comments to make sure this assumption is not missed.

Reply:
Thanks!